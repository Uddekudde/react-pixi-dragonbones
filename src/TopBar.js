import React from 'react';
import { Link } from 'react-router-dom';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';

import AnimationDrawer from './AnimationDrawer.js';

export default class TopBar extends React.Component {

    render(){
        return(
            <AppBar color="default" style={{flexGrow: 1, background: 'transparent', boxShadow: 'none'}} >
                <Toolbar >
                    <Typography variant="title" color="inherit">
                        {/*<Link to='/'>Home</Link>
                        <Link to='/pixi'>アニメ絵ション</Link>*/}
                        <Button component={Link} style={{color: 'white'}} to='/about'>
                            About
                        </Button>
                    </Typography>
                    <div style={{flex: '1 100px'}}>
                    </div>
                    <AnimationDrawer/>
                </Toolbar>
            </AppBar>
        )
    }
}