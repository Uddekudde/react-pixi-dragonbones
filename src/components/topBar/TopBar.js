import React from 'react';
import { Link } from 'react-router-dom';
import { HashRouter } from 'react-router-dom';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';

//import AnimationDrawerContainer from '../animationDrawer/AnimationDrawerContainer.js';

export default class TopBar extends React.Component {

    render(){
        return(
            <HashRouter>
            <AppBar color="default" style={{flexGrow: 1, background: 'transparent', boxShadow: 'none'}} >
                <Toolbar >
                    <Typography variant="title" color="inherit">
                        <Button component={Link} style={{color: 'white'}} to='/about' onClick={() => this.props.overlayActive('true')}>
                            About
                        </Button>
                    </Typography>
                    <div style={{flex: '1 100px'}}>
                    </div>
                    {/*<AnimationDrawerContainer/>*/}
                </Toolbar>
            </AppBar>
            </HashRouter>
        )
    }
}