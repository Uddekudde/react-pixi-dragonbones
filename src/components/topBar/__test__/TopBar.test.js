import React from 'react';
import ReactDOM from 'react-dom';
import '@testing-library/jest-dom/extend-expect';
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import TopBar from "../TopBar";

afterEach(cleanup);
it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<TopBar/>, div);
});

it("renders button correctly", () => {
    const { getByText } = render(<TopBar />);
    expect(getByText('About')).toBeInTheDocument();
});

it("matches snapshot", () => {
    const tree = renderer.create(<TopBar />).toJSON();
    expect(tree).toMatchSnapshot();
});

