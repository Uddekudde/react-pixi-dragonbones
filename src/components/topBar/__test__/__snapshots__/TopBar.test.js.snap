// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<header
  className="MuiPaper-root-9 MuiPaper-elevation4-15 MuiAppBar-root-1 MuiAppBar-positionFixed-2 MuiAppBar-colorDefault-6 mui-fixed"
  style={
    Object {
      "background": "transparent",
      "boxShadow": "none",
      "flexGrow": 1,
    }
  }
>
  <div
    className="MuiToolbar-root-36 MuiToolbar-regular-38 MuiToolbar-gutters-37"
  >
    <h2
      className="MuiTypography-root-40 MuiTypography-title-46 MuiTypography-colorInherit-59"
    >
      <a
        className="MuiButtonBase-root-91 MuiButton-root-65 MuiButton-text-67 MuiButton-flat-70"
        href="#/about"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        style={
          Object {
            "color": "white",
          }
        }
        tabIndex="0"
      >
        <span
          className="MuiButton-label-66"
        >
          About
        </span>
        <span
          className="MuiTouchRipple-root-94"
        />
      </a>
    </h2>
    <div
      style={
        Object {
          "flex": "1 100px",
        }
      }
    />
  </div>
</header>
`;
